core-bundle/src/Doctrine/Backup/BackupManager.php

```diff
@@ -0,0 +1,311 @@
+     */
+    private function doRestore(RestoreConfig $config): void
+    {
+        $backup = $config->getBackup();
+
+        if (!file_exists($backup->getFilepath())) {
+            throw new BackupManagerException(sprintf('Dump does not exist at "%s".', $backup->getFilepath()));
+        }
+
+        $handle = $config->isGzCompressionEnabled() ? gzopen($backup->getFilepath(), 'r') : fopen($backup->getFilePath(), 'r');
+
+        $currentQuery = '';
+        $checkedForHeader = $config->ignoreOriginCheck();
+
+        while ($line = $config->isGzCompressionEnabled() ? gzgets($handle) : fgets($handle)) {
+            $line = trim($line);
+
+            if (!$checkedForHeader) {
+                if ($line !== $this->getDumpHeader()) {
+                    throw new BackupManagerException('The Contao database importer only supports dumps generated by Contao.');
+                }
+                $checkedForHeader = true;
+                continue;
+            }
+
+            // Ignore comments
+            if ('--' === substr($line, 0, 2)) {
+                continue;
+            }
+
+            $currentQuery .= $line;
+
+            // Current query ends
+            if (';' === substr($currentQuery, -1)) {
+                $this->executeWrappedQuery($currentQuery);
+                $currentQuery = '';
+            }
+        }
+
+        if ('' !== $currentQuery) {
+            $this->executeWrappedQuery($currentQuery);
+        }
+    }
+
+    private function tidyDirectory(): void
+    {
+        $i = 0;
+
+        foreach ($this->listBackups() as $backup) {
+            if ($i >= $this->keepMax) {
+                (new Filesystem())->remove($backup->getFilepath());
+            }
+            ++$i;
+        }
+    }
+
+    private function getDumpHeader(): string
+    {
+        return sprintf(self::DUMP_HEADER, self::DUMP_VERSION);
+    }
+
+    private function createSlimDumpConfig(AbstractConfig $config): Config
+    {
+        $tables = $this->connection->createSchemaManager()->listTables();
+        $tableNames = array_map(static fn (Table $table) => $table->getName(), $tables);
+
+        $tableNames = array_diff($tableNames, $config->getTablesToIgnore());
+
+        $doc = new \DOMDocument();
+        $slimDump = $doc->createElement('slimdump');
+
+        foreach ($tableNames as $tableName) {
+            $table = $doc->createElement('table');
+            $table->setAttribute('name', $tableName);
+            $table->setAttribute('dump', 'full');
+            $slimDump->appendChild($table);
+        }
+
+        $doc->appendChild($slimDump);
+
+        return ConfigBuilder::createFromXmlString($doc->saveXML());
+    }
+
+    /**
+     * @throws BackupManagerException
+     */
+    private function executeWrappedQuery(string $query): void
+    {
+        try {
+            $this->connection->executeQuery($query);
+        } catch (Exception $e) {
+            throw new BackupManagerException($e->getMessage());
```

Comment:
should probably pass `$previous` (applies to other places as well)

Reply:
Done in f2a2fcbaca9ddc1b1ccda92e57319e728667927f.